### PR TITLE
Lesson retirement policy

### PR DESCRIPTION
**Pull request by mdlincoln**
This PR will show a preview of the Lesson Retirement Policy page draft re: #536 

Discussion of exact language and tasks will take place here. Any additions can be pushed to the `archiving-policy` branch.

Preview: https://deploy-preview-612--ph-dev.netlify.com/lesson-retirement-policy

Preview lesson: https://deploy-preview-612--ph-dev.netlify.com/lessons/retired/counting-frequencies-from-zotero-items

To-do:

- [x] reinstate `translation-editor` field checking before merging, temporarily removed so site will properly build (re: #611) ~(do by reverting 318b712bc23c987b48d0490e65146636c5bdb6d2)~ **already completed and merged to `gh-pages`**
- [x] rename `lessons/deprecated` to `lessons/retired`, add appropriate redirects
- [x] Language about twitter bot #514
- [x] Mention sustainability goals #534 
- [x] Spanish translations
    - [x] New page on retirement policy 
    - [x] a few new lines in [_data/snippets.yml](https://github.com/programminghistorian/jekyll/pull/612/files#diff-a407e023d60dfefccac66d5a7959003c)
    - [x] additions to [author guidelines](https://github.com/programminghistorian/jekyll/pull/612/files#diff-72a7d8f7b711a8af34fe8532ab83c3a6)
    - [x] additions to [editor guidelines](https://github.com/programminghistorian/jekyll/pull/612/files#diff-542dbfd70101e87b830b47847632b7c1)
    - [x] additions to [reviewer guidelines](https://github.com/programminghistorian/jekyll/pull/612/files#diff-a038c8ad532ac4704b06f260e545d023)
- [x] List retired lessons
- [x] Fix display of images on retired lessons
- [x] add `retired_date` to retired lessons

### CODE REVIEW

**Comment by drjwbaker** [comment]
On the 'announcement' call out, I think this should start more bluntly. Perhaps:

> **This lesson has been retired**
> *What does this mean*: The Programming Historian editors do their best to maintain lessons as minor issues inevitably arise. However, since publication, changes to either the underlying technologies or principles used by this lesson have been substantial, to the point where the editors have decided not to further update it. The lesson may still prove a useful learning tool and a snapshot into the techniques of digital history when it was published, but we cannot guarantee all elements will continue to work as intended.

**Comment by acrymble** [comment]
I think this is great.

**Comment by drjwbaker** [comment]
On https://github.com/programminghistorian/jekyll/commit/1ae452f I know this is an 'archive policy' but I have a problem with the wording "This lesson has been archived". Imagine you are a user who has no knowledge of our work or this policy. One very reasonable reading of "This lesson has been archived" is not that it has been deprecated/retired/[similar] but that we are archiving the lesson in a positive preservation sense. So whilst I'm neither wedded by my "This lesson has been retired" wording nor wish to change the policy from "Archiving policy" (because it works), if we have a bolder statement as I think we should https://github.com/programminghistorian/jekyll/pull/612#issuecomment-333459888 we need to think carefully about the wording.

Possible options:

- "This lesson has been archived"
- "This lesson has been retired"
- "This lesson is no longer maintained"
- "This lesson is no longer functional"
- "This lesson is no longer maintained|functional and has been archived|retired"

Sorry for being fussy on this one!

**Comment by mdlincoln** [comment]
No apologies necessary @drjwbaker - wording is everything (or at least 90% of things) here. 

I quite like the "retired" suggestion, as it's not as negatively charged as "deprecated" but also indicates something that is no longer quite current, in a way that "archived" doesn't necessarily do.

I _would_ like to keep the wording consistent, meaning that we discuss lessons as "retired", "retirement policy", create `/lessons/retired/TITLE` URLs, etc. Do we have strong feelings about using the term "retired" instead?

**Comment by alsalin** [comment]
I think the statement is great! Fwiw, I don't mind the "archive" language.

**Comment by jerielizabeth** [comment]
I just spent a few minutes googling the different ideas and my (very unscientific and probably shaped by previous browsing) impression is that "archive" does have connotations of preservation and being pulled out of circulation, which is not quite what we're doing here. Software documentation and Software Carpentry use "previous" and "previous versions," though those are more clearly versioned and so don't quite fit.

I found a few analogous uses of "retired software" so that would have good resonance. I agree with @mdlincoln that "retired" is much softer than "deprecated" but I am not sure about the "Retirement Policy" idea :D I do, however, like the other uses of "retired" in the URL and such. 

So ... all that to say, I'm team "retired" in all things except the Policy name.

**Comment by mdlincoln** [comment]
That's really useful info @jerielizabeth. Is there a reason to not call it the "retirement policy"?

**Comment by jerielizabeth** [comment]
@mdlincoln So further thinking with additional coffee ... What about "Lesson Retirement Policy"? I think my gut reaction is b/c I associate "Retirement Policy" with humans, not objects.

**Comment by ianmilligan1** [comment]
Thanks for this @mdlincoln!

After reading the discussion, I think the "Lesson Retirement Policy" works well, and that a banner statement of "This lesson has been retired" with a link to the policy doesn't have the negative connotation of "depreciated" nor the fuzziness of "archive."

I wonder if it would be possible to have a page somewhere (it can be a few clicks away from the main page) with our retired lessons? While they shouldn't be on the active page, somebody might want to explore them to see the sorts of lessons we have published in the past, and as a way to reflect on the evolution of the field.

File: archiving-policy.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+---
+title: Archiving Policy
+layout: blank
+---
+
+# Archiving Policy
+
+The Programming Historian editors do their best to maintain lessons as minor issues inevitably arise.
```
Comment (ianmilligan1) [inline archiving-policy.md]: For consistency, should this be the _Programming Historian_?

**Comment by mdlincoln** [comment]
I agree it's probably fair to have a list of the lessons - what do you think of simply linking to such a list from the retirement policy page? That way someone coming across them has seen their context?

**Comment by acrymble** [comment]
rather than a new page, why not just add a filter button to the main lessons page (show/hide retired lessons)?

**Comment by drjwbaker** [comment]
Looking good @mdlincoln :)

**Comment by ianmilligan1** [comment]
I can see merits in both including on main page and in the lessons retirement policy... I guess I worry our lessons page already is quite active, so adding more might confuse more than it adds. My vote would be to include a list on the retirement page.

**Comment by mdlincoln** [comment]
Any suggestions about changes to the display of retired lessons themselves? @drjwbaker I remember you had some thoughts swirling around on the last call.

(let's not go crazy, however - we have a limited amount of metadata for these lessons)

Example retired lesson: https://deploy-preview-612--ph-dev.netlify.com/lessons/retired/counting-frequencies-from-zotero-items

**Comment by ianmilligan1** [comment]
FWIW I think the example text you have looks great, @mdlincoln – is it trivial to make a header like that a "sticky header," in case somebody comes into the lesson midway through or moves past the notice too quickly? (if it is non-trivial, I don't think it's worth a substantial investment of resources or time)

**Comment by acrymble** [comment]
In keeping with our visual rhetoric, why not add an image of something
metaphorically relevant, like the lesson icons. That might make it more
eye-catching and less likely to be missed.

https://www.flickr.com/photos/britishlibrary/11024506983/

**Comment by walshbr** [comment]
Sorry for getting to this late - great work! This looks great to me, particularly all the changes that you've all suggested. Retirement sounds like a good way to describe it. And I also like Adam's idea of including the lesson icons with the list of retired lessons. 

Also appreciated "twitter announcement stream" as the kindest way I've ever heard a twitter bot described.

**Comment by drjwbaker** [comment]
@acrymble Nice use of those 1m images Adam :)

**Comment by drjwbaker** [comment]
@mdlincoln My suggestions were:
- change 'modified' date in the header to 'retired' date.
- change the colouring so the look/feel of the page is different to normal PH
- link modified date to last annual dump in which lesson appears (once that is up and running)

**Comment by mdlincoln** [comment]
@drjwbaker yes, having a `retired-date` field makes sense.

I don't understand at all what you mean by linking to the last Zenodo version? I don't know what is gained there, or how that interacts with the `retired-date` field?

**Comment by drjwbaker** [comment]
> I don't understand at all what you mean by linking to the last Zenodo version? I don't know what is gained there, or how that interacts with the retired-date field?

Ignore me  on that one :)

**Comment by mdlincoln** [comment]
@alsalin thanks for pushing your changes onto this PR! I added to the Retirement Policy page some cross-references to the sustainability segments of the author, editor, and review guidelines. They all look consistent to me.

Are there any further edits to text from anyone else on the team? The next step would be to have the Spanish team add their translations, so we should be careful to settle on our copy before doing that.

**Comment by acrymble** [comment]
I think it's worth sending this out to the collective via some tweets to get feedback. Sometimes we get good ideas from the crowd. That saves us running into things we didn't think of later, and needing the Spanish team to react.

**Comment by drjwbaker** [comment]
Is the plan to connect this and the author guidelines up in some way? https://programminghistorian.org/author-guidelines (in addition to connecting this and the sustainable writing stuff @alsalin has been working on)

**Comment by mdlincoln** [comment]
@drjwbaker see above - @alsalin has added those changes to this PR, and I've made links from the retirement policy to the sustainability sections of each guidelines page: https://deploy-preview-612--ph-dev.netlify.com/lesson-retirement-policy#related-sustainability-guidelines

**Comment by drjwbaker** [comment]
@mdlincoln In that case - and for the second time recently - ignore me :)

**Comment by taylor-arnold** [comment]
Sorry, just saw this via Twitter (thanks @acrymble for thinking of those of us in the "collective"!).

The language around this seems quite reasonable. It seems to split the difference between stopping readers from getting stuck on outdated posts while not being overly negative to the authors of the original piece.

Looking at the example archived articles, it would be nice if there was some way (perhaps a metadata field and displayed below the warning) that the editors could explain specifically why the tutorial has been retired. I suspect in most cases it is something like *the underlying tool X is no longer supported* or *the web resource Y is no longer available*. In tricky situations it could always be left blank. This seems like a useful feature because it further makes it clear that retirement is not a mark against the original author. It also provides some documentation for curious readers and fits in with the open peer-review model of PH.

**Comment by mdlincoln** [comment]
@statsmaths That's an excellent suggestion - we can certainly include some more custom narrative on the reasons for retirement. In those short notes, perhaps we might also link to relevant tickets on the repo where discussion took place.

I can try to mock this up for e.g. the Zotero lessons and we can see how it looks.

**Comment by mdlincoln** [comment]
Here's an example of a reasoned retirement alert: https://deploy-preview-612--ph-dev.netlify.com/lessons/retired/counting-frequencies-from-zotero-items

(This is just an option in the metadata - without a specific reason, it'll just show the warning.)

**Comment by drjwbaker** [comment]
Looks great. Linking to the relevant Issue was exactly what I had in mind when I read @statsmaths's comment.

**Comment by taylor-arnold** [comment]
Looks great. I hadn't realized there would be an issue attached to each retirement decision. Linking to that is perfect. Thanks!

**Comment by drjwbaker** [comment]
From the collective https://twitter.com/Ashenground/status/920431841764626432

> Will you add a link through from a retired lesson to its replacement(s) + add a specific retirement reason?

**Comment by mdlincoln** [comment]
Who on the Spanish team is taking responsibility for translating this?

**Comment by mdlincoln** [comment]
@arojascastro @mariajoafana @vgayolrs any updates on this?

**Comment by arojascastro** [comment]
Hey @mdlincoln, sorry I did not see this. I promise to translate on the weekend.

**Comment by arojascastro** [comment]
Ok, here goes my translation - I'll open an issue because I have some doubts.

Los editores de *The Programming Historian* hacemos todo lo posible para mantener el funcionamiento de las lecciones a medida que surgen problemas menores. Sin embargo, con el paso del tiempo, los cambios en las tecnologías subyacentes o los principios utilizados por una lección determinada pueden llegar a ser tan sustanciales que los usuarios ya no podrán completar la lección con éxito. Ante estas situaciones, el equipo editorial del *The Programming Historian en español* puede decidir no traducir una lección (si el equipo de editores inglés ya la ha retirado) o bien, si ya ha sido traducida, "retirar" una lección: mantener la página publicada, pero eliminarla de nuestro directorio de lecciones activas y añadir una advertencia en la parte superior de la página señalando que no todos los elementos de la lección pueden estar funcionando como se pretendía originalmente. Esta última medida también se llevaría a cabo con las lecciones originales en español.

No retiramos las lecciones a la ligera. Si el equipo editorial comienza a recibir mensajes sobre problemas con una lección, investigaremos el problema y tomaremos medidas para resolverlo. A menudo, decidiremos que no es necesario retirar la lección:

- No retiraremos la lección cuando las correcciones son fáciles de hacer (por ejemplo, corregir el formato de una lección o cambiar una o dos URLs rotas).
- No retiraremos la lección cuando han surgido nuevos métodos para una tarea preferibles a los métodos expuestos en una lección, pero la tecnología original para una lección todavía funciona y está disponible. La lección aún puede resultar una herramienta de aprendizaje útil y proporcionar una instantánea de las técnicas disponbiles cuando se publicó.

Sin embargo, si está claro que los cambios en el texto, código y/o figuras son necesarios de manera sistemática y que tales cambios son de una escala que requeriría rehacer la lección por completo, entonces abriremos un debate público para discutir entre todos los miembros del equipo editorial la posibilidad de retirar la lección.

En los casos en que los miembros del equipo editorial, o miembros de la comunidad en general, estén dispuestos y sean capaces de ofrecer voluntariamente su experiencia, podemos crear una lección actualizada derivada del original. De acuerdo con nuestra licencia CC-BY, este derivado atribuirá visiblemente al creador de la lección original, así como los nombres de todos los colaboradores que contribuyeron en la producción de la nueva lección.

Con independencia de que se cree o no un nuevo derivado, llevaremos a cabo los siguientes pasos con la lección retirada:

- La lección será trasladada de https://programminghistorian.org/es/lecciones/TITULO-LECCION a https://programminghistorian.org/es/lecciones/retiradas/TITULO-LECCION. Se establecerá una redirección, por lo que cualquier enlace a la URL original indicará al usuario la nueva URL.

- Una vez retirada, la lección dejará de aparecer en el directorio de lecciones y se eliminará del flujo de anuncios de Twitter.

- El siguiente anuncio será añadido a la parte superior de la lección retirada:

>Los editores de *The Programming Historian* hacen todo lo posible para mantener las lecciones a medida que surgen inevitablemente problemas menores. Sin embargo, desde la publicación, los cambios en las tecnologías subyacentes o los principios utilizados en esta lección han sido sustanciales, hasta el punto de que los editores han decidido no actualizarla más. Es posible que la lección siga siendo una herramienta de aprendizaje útil y una instantánea de las técnicas de la historia digital cuando se publicó, pero no podemos garantizar que todos los elementos sigan funcionando como estaba previsto.

## Más sobre sostenibilidad 

## Lecciones retiradas

**Comment by arojascastro** [comment]
I have uploaded my translation in committ  3a5f44d

**Comment by mdlincoln** [comment]
Thanks @arojascastro for taking on the translation of the brand new page. There are still a few translation tasks left on existing pages:

- a few new lines in [_data/snippets.yml](https://github.com/programminghistorian/jekyll/pull/612/files#diff-a407e023d60dfefccac66d5a7959003c)
- additions to [author guidelines](https://github.com/programminghistorian/jekyll/pull/612/files#diff-72a7d8f7b711a8af34fe8532ab83c3a6)
- additions to [editor guidelines](https://github.com/programminghistorian/jekyll/pull/612/files#diff-542dbfd70101e87b830b47847632b7c1)
- additions to [reviewer guidelines](https://github.com/programminghistorian/jekyll/pull/612/files#diff-a038c8ad532ac4704b06f260e545d023)

You can view all the changes in the pull request here https://github.com/programminghistorian/jekyll/pull/612/files

**Comment by mdlincoln** [comment]
(and I'm sorry there's so much text to translate 🤕  This was a very big policy update!)

**Comment by arojascastro** [comment]
I think I am done with the translations... it would be good that after merging and publishing, our colleagues @jenniferisasi and @jamotilla revised all the guidelines that I updated.

**Comment by mdlincoln** [comment]
Travis is backlogged this morning, so it's checks are still in-progress - but I've run the checks locally and it works fine! If @ianmilligan1 or @walshbr or someone could give their review approval, I think we can finally merge this! 🎉

**Review by walshbr — APPROVED** [review]
Looks good to me! 👍 Checks have passed now, and the pages are coming up as expected as far as I can tell.

**Pull request merged**